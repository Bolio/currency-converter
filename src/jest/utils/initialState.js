const initialState = {
  currency: {
    currentConversion: {},
    fetching: false,
    success: false,
    error: false,
    errorMessage: "",
  },
};

export default initialState;
