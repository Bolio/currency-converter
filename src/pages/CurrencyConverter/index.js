import React from "react";
import Container from "@mui/material/Container";
import Paper from "@mui/material/Paper";
import Divider from "@mui/material/Divider";

function CurrencyConverter() {
  return (
    <Container maxWidth="lg" sx={{ mt: 4, mb: 4 }}>
      <Paper sx={{ p: 2, display: "flex", flexDirection: "column" }}>
        <MainTitle
          text={"Currency Converter 🏦"}
          align={"center"}
          variant={"h3"}
          gutterBottom={true}
        />
        <Divider />
      </Paper>
    </Container>
  );
}

export default CurrencyConverter;
