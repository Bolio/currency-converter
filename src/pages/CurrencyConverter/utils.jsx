import React from "react";
import CompareArrowsIcon from "@mui/icons-material/CompareArrows";
import ArrowBackIcon from "@mui/icons-material/ArrowBack";
import ArrowForwardIcon from "@mui/icons-material/ArrowForward";

const choiceCurrency = (choice) => {
  if (choice === undefined) {
    return <CompareArrowsIcon sx={{ fontSize: 60 }} />;
  } else {
    return choice === "USD" ? (
      <ArrowForwardIcon sx={{ fontSize: 60 }} />
    ) : (
      <ArrowBackIcon sx={{ fontSize: 60 }} />
    );
  }
};

export { choiceCurrency };
