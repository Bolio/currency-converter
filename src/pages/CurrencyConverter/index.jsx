import React, { useState, useEffect } from "react";
import { useDispatch } from "react-redux";
import Container from "@mui/material/Container";
import Paper from "@mui/material/Paper";
import Grid from "@mui/material/Grid";

import MainTitle from "../../components/title/MainTitle";
import Input from "../../components/Input";
import { choiceCurrency } from "./utils";
import { fetchInformationCurrenciesThunk } from "../../redux/modules/currencyConverter";

function CurrencyConverter() {
  const dispatch = useDispatch();
  const [choice, setChoice] = useState(undefined);
  const [valueMxn, setValueMxn] = useState("");
  const [valueUsd, setValueUsd] = useState("");

  useEffect(() => {
    console.log("useEffect");
    dispatch(fetchInformationCurrenciesThunk());
  }, []);

  const handleChangeInput = (value, currency) => {
    // console.log("currency:", currency);
    // console.log("value:", value);

    if (currency === "USD") {
      setValueUsd(value);
      setValueMxn(value * 2);
    } else if (currency === "MXN") {
      setValueMxn(value);
      setValueUsd(value * 4);
    }
  };

  const handleFocus = (choice) => {
    setChoice(choice);
    setValueMxn("");
    setValueUsd("");
  };

  return (
    <Container maxWidth="lg" sx={{ mt: 4, mb: 4 }}>
      <Paper sx={{ p: 2, display: "flex", flexDirection: "column" }}>
        <MainTitle
          text={"Currency Converter ðŸ’±"}
          align={"center"}
          variant={"h3"}
          gutterBottom={true}
        />
        <Grid container>
          <Grid item xs={5}>
            <Input
              id="currency-usd"
              label="USD"
              type="number"
              focus={(e) => handleFocus("USD")}
              change={(e) => handleChangeInput(e.target.value, "USD")}
              value={valueUsd}
            />
          </Grid>
          <Grid item xs={2} sx={{ textAlign: "center" }}>
            {choiceCurrency(choice)}
          </Grid>
          <Grid item xs={5}>
            <Input
              id="curency-mxn"
              label="MXN"
              type="number"
              focus={(e) => handleFocus("MXN")}
              change={(e) => handleChangeInput(e.target.value, "MXN")}
              value={valueMxn}
            />
          </Grid>
        </Grid>
      </Paper>
    </Container>
  );
}

export default CurrencyConverter;
