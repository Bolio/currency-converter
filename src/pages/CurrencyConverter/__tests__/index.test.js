import React from "react";
import configureMockStore from "redux-mock-store";
import CurrencyConverter from "..";
import { render } from "@testing-library/react";
import initialState from "../../../jest/utils/initialState";
import { renderWithProviderSnapshot } from "../../../jest/utils";

const mockStore = configureMockStore([]);

console.log("initialState", initialState);

const currentConversionInitialState = {
  ...initialState,
  currency: {
    currentConversion: {
      date: "2022-05-19",
      info: {
        rate: 19.92868,
        timestamp: 1652997843,
      },
      query: {
        amount: 1,
        from: "USD",
        to: "MXN",
      },
      result: 19.92868,
      success: true,
    },
    fetching: false,
    success: false,
    error: false,
    errorMessage: "",
  },
};
const store = mockStore(currentConversionInitialState);

describe("CurrencyConverter component", () => {
  const component = renderWithProviderSnapshot(<CurrencyConverter />, {
    initialState: currentConversionInitialState,
    store,
  });

  test("Should render CurrencyConverter component", () => {
    const container = component;
    expect(container).toMatchSnapshot();
  });
});
