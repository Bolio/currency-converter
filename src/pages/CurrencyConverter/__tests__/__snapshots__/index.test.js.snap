// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyConverter component Should render CurrencyConverter component 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1f2kw6w-MuiContainer-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1slnj5u-MuiPaper-root"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter MuiTypography-gutterBottom css-1le49ku-MuiTypography-root"
    >
      Currency Converter ðŸ’±
    </h3>
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="currency-usd"
            id="currency-usd-label"
          >
            USD
          </label>
          <div
            className="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-gjbq6i-MuiInputBase-root-MuiFilledInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
              disabled={false}
              id="currency-usd"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1thszbi-MuiGrid-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1dxcdol-MuiSvgIcon-root"
          data-testid="CompareArrowsIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9.01 14H2v2h7.01v3L13 15l-3.99-4v3zm5.98-1v-3H22V8h-7.01V5L11 9l3.99 4z"
          />
        </svg>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="curency-mxn"
            id="curency-mxn-label"
          >
            MXN
          </label>
          <div
            className="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-gjbq6i-MuiInputBase-root-MuiFilledInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
              disabled={false}
              id="curency-mxn"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
